src/extension/infer.rs

```diff
@@ -279,7 +279,13 @@ impl UnificationContext {
                     let m_input_node = self.make_or_get_meta(input, dir);
                     self.add_constraint(m_input_node, Constraint::Equal(m_input));
                     let m_output_node = self.make_or_get_meta(output, dir);
-                    self.add_constraint(m_output_node, Constraint::Equal(m_output));
+                    // If the parent node is a FuncDefn, it's input_resources
+                    // should always be empty, meaning if the function adds
+                    // resources, they shouldn't be equal to the parent's output
+                    // (which will also be empty.)
```

Comment:
Is that true? parent's output is equal to (its input, probably empty, maybe not necessarily) plus delta

Reply:
Sorry, yes, well, the "its input_extensions should always be empty" bit is as above, but the point to make is that the parent's output is equal to its input, because these do not include the delta of the function itself.

Reply:
Removed the stuff about empty input_extensions, since it isn't true